### ISSUE TITLE
Implement Key Metrics for Watcher Listener Service

### ISSUE THREAD

**Issue by ruanxin** [kyma-project/lifecycle-manager#750]
# Description:
As part of enhancing the performance insights for the Watcher Listener service, we need to implement various metrics to track its performance and behavior. The following metrics should be implemented:

**RED Metrics:**
1. Request Rate: The number of requests per second that the service is serving.
1. Request Errors: The number of failed requests per second.
1. Request Duration: Distribution of the amount of time each request takes.

**Additional Metrics:**
1. Inflight Requests: Records the number of requests being handled concurrently at any given time (with the handler).
1. Count of Requests Exceeding Size Limit: Keeps track of the number of requests that exceed a certain size limit.
1. Count of Requests Not Being Verified: Keeps track of the number of requests that failed verification

For implementation, we can refer to the `go-http-metrics` library as an example:
[go-http-metrics middleware](https://github.com/slok/go-http-metrics/blob/master/middleware/middleware.go#L67-L119)

# References:
- [The RED Method: How to Instrument Your Services](https://grafana.com/blog/2018/08/02/the-red-method-how-to-instrument-your-services/)
- [The RED Method - Key Metrics for Microservices Architecture](https://www.weave.works/blog/the-red-method-key-metrics-for-microservices-architecture/)

# Acceptance Criteria:
- [x] Implement all the mentioned metrics in the `http.handler` function of the Watcher Listener service.
- [ ] Metrics should be exposed through the existing metrics endpoint.
- [ ] Create a follow-up issue to add these metrics to the Grafana dashboard for visualization.

**Comment by janmedrek** [comment]
Blocked by https://github.com/kyma-project/runtime-watcher/issues/115

**Comment by nesmabadr** [comment]
Followup issue to add the metrics to Grafana dashboard: https://github.com/kyma-project/lifecycle-manager/issues/830
